app/models/alchemy/element.rb

```diff
@@ -321,9 +321,11 @@ def copy_nested_elements_to(target_element)
     #   Pass an element name to get previous of this kind.
     #
     def previous_or_next(dir, name = nil)
-      elements = page.elements.published.where("#{self.class.table_name}.position #{dir} #{position}")
+      dir = dir == '>' ? '>' : '<'
+      order = dir == '>' ? :asc : :desc
```

Comment:
Hi @tvdeyen, thanks for the review! I'd be happy to make some changes to this PR, but just a thought here:

If line 324 is removed, then the possibility for SQL injection remains via a malicious `dir` variable.

For example, imagine calling the method like this:

```ruby
previous_or_next('< 1); DROP TABLE alchemy_pages; --', name)
```

Line 324 is present to ensure that this can never occur (even though the method is private).

My initial thought was that maybe we could refactor the `previous_or_next` method into separate `previous` and `next` methods so that interpolation in the query would not be needed.

Would this be something that you're open to?